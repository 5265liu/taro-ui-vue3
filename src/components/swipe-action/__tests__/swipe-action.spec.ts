import { mount } from '@vue/test-utils'
import { h, VNode } from '@vue/runtime-core'

import AtSwipeAction from '../index'

const factory = (
  values = {},
  slots: { default: Array<VNode | string> } = { default: [''] }
) => {
  return mount(AtSwipeAction as any, {
    slots,
    props: { ...values },
  })
}

const MAX_OFFSET_SIZE = 101

const OPTIONS = [
  {
    text: '取消',
    class: 'cancel',
    style: {
      backgroundColor: '#6190E8',
    },
  },
  {
    text: '确认',
    class: 'confirm',
    style: {
      backgroundColor: '#FF4949',
    },
  },
]

const DOM_INFO = {
  top: 1,
  bottom: 44, // top + height
  left: 0,
  right: 375, // left + width
  height: 43,
  width: 375,
}

const START_INFO = { clientX: 0, clientY: 0 }
const MOVE_INFO = {
  clientY: 0,
  clientX: -(MAX_OFFSET_SIZE - 1),
  pageX: DOM_INFO.width / 2,
  pageY: DOM_INFO.height / 2,
}

describe('SwipeAction Snap', () => {
  it('should render SwipeAction -- props options', () => {
    const wrapper = mount({
      render() {
        return (
          h(AtSwipeAction, {
            disabled: true,
            autoClose: true,
            options: OPTIONS,
            class: "swipe-action--test"
          }, {
            default: () => [
              h('view', { class: "normal" }, "AtSwipeAction 一般使用场景")
            ]
          })
        )
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render SwipeAction -- isOpened', () => {
    const wrapper = mount({
      render() {
        return (
          h(AtSwipeAction, {
            isOpened: true,
            disabled: true,
            autoClose: true,
            options: OPTIONS,
            class: "swipe-action--test"
          }, {
            default: () => [
              h('view', { class: "normal" }, "AtSwipeAction 一般使用场景")
            ]
          })
        )
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
