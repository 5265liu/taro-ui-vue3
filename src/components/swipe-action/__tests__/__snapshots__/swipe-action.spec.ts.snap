// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeAction Snap should render SwipeAction -- isOpened 1`] = `
<view
  class="swipe-action--test at-swipe-action swipe-action--test"
  id="swipeAction-NPKDAANB"
>
  <view
    class="at-swipe-action__content animation"
  >
    <view
      class="normal"
    >
      AtSwipeAction 一般使用场景
    </view>
  </view>
  <view
    class="at-swipe-action__options"
    id="swipeActionOptions-NPKDAANB"
  >
    <view
      class="at-swipe-action__option"
      style="background-color: rgb(97, 144, 232);"
    >
      <text
        class="option__text"
      >
        取消
      </text>
    </view>
    <view
      class="at-swipe-action__option"
      style="background-color: rgb(255, 73, 73);"
    >
      <text
        class="option__text"
      >
        确认
      </text>
    </view>
  </view>
</view>
`;

exports[`SwipeAction Snap should render SwipeAction -- props options 1`] = `
<view
  class="swipe-action--test at-swipe-action swipe-action--test"
  id="swipeAction-DOKLFONB"
>
  <view
    class="at-swipe-action__content animation"
  >
    <view
      class="normal"
    >
      AtSwipeAction 一般使用场景
    </view>
  </view>
  <view
    class="at-swipe-action__options"
    id="swipeActionOptions-DOKLFONB"
  >
    <view
      class="at-swipe-action__option"
      style="background-color: rgb(97, 144, 232);"
    >
      <text
        class="option__text"
      >
        取消
      </text>
    </view>
    <view
      class="at-swipe-action__option"
      style="background-color: rgb(255, 73, 73);"
    >
      <text
        class="option__text"
      >
        确认
      </text>
    </view>
  </view>
</view>
`;
